api/authenticator.py

```diff
@@ -416,6 +433,10 @@ def __init__(self, library, basic_auth_provider=None,
                 self.oauth_providers_by_name[provider.NAME] = provider
         self.assert_ready_for_oauth()
 
+    @property
+    def library(self):
+        return get_one(self._db, Library, id=self.library_id)
```

Comment:
I think you're right, it looks like library_id is only passed on to the Authenticators. I'll rerun the full test suite to be sure.